src/bun.js/bindings/Script.cpp

```diff
@@ -0,0 +1,181 @@
+#include "root.h"
+
+#include "Script.h"
+#include "JavaScriptCore/JSObjectInlines.h"
+#include "wtf/text/ExternalStringImpl.h"
+
+#include "JavaScriptCore/FunctionPrototype.h"
+#include "JavaScriptCore/HeapAnalyzer.h"
+
+#include "JavaScriptCore/JSDestructibleObjectHeapCellType.h"
+#include "JavaScriptCore/SlotVisitorMacros.h"
+#include "JavaScriptCore/ObjectConstructor.h"
+#include "JavaScriptCore/SubspaceInlines.h"
+#include "wtf/GetPtr.h"
+#include "wtf/PointerPreparations.h"
+#include "wtf/URL.h"
+#include "JavaScriptCore/TypedArrayInlines.h"
+#include "JavaScriptCore/PropertyNameArray.h"
+#include "Buffer.h"
+#include "GCDefferalContext.h"
+#include "Buffer.h"
+
+#include <JavaScriptCore/DOMJITAbstractHeap.h>
+#include "DOMJITIDLConvert.h"
+#include "DOMJITIDLType.h"
+#include "DOMJITIDLTypeFilter.h"
+#include "DOMJITHelpers.h"
+#include <JavaScriptCore/DFGAbstractHeap.h>
+
+namespace WebCore {
+using namespace JSC;
+
+static EncodedJSValue constructScript(JSGlobalObject* globalObject, CallFrame* callFrame, JSValue newTarget = JSValue())
+{
+    VM& vm = globalObject->vm();
+    JSValue callee = callFrame->jsCallee();
+    ArgList args(callFrame);
+    JSValue sourceArg = args.at(0);
+    String source = sourceArg.isUndefined() ? emptyString() : sourceArg.toWTFString(globalObject);
+
+    Structure* structure = Script::createStructure(vm, globalObject, Script::createPrototype(vm, globalObject));
```

Comment:
This one we will want to cache lazily on the global object

`LazyProperty<JSGlobalObject, Structure>`

That will make this faster at runtime to create these

Reply:
I've addressed this using a `LazyClassStructure`, which should cache the structure, prototype, and constructor all in one :)